batch/batches/Convert/Engines/KJobConversionEngine.class.php

```diff
@@ -146,13 +147,65 @@ public function convertJob ( KalturaConvertJobData &$data )
 		
 		return array ( true , $error_message );// indicate all was converted properly
 	}
-	
+
+	protected function isConversionProgressing($currentModificationTime)
+	{
+		if (is_file($this->inFilePath))
+		{
+			$newModificationTime = filemtime($this->inFilePath);
+			if ($newModificationTime !== false && $newModificationTime > $currentModificationTime)
+			{
+				return $newModificationTime;
+			}
+		}
+		return false;
+	}
+
+	protected function getReturnValues($handle)
+	{
+		$return_var = false;
+		$output = false;
+		if ($handle)
+		{
+			$return_var = 0;
+			$file = escapeshellarg($this->outFilePath);
+			if (is_file($file))
+			{
+				$output = `tail -n 1 $file`;
```

Comment:
instead of back-ticks , use kFile::getFileTail ($fileName , $numOFLines)l;

If function does not exist - add it